bool ignore
---
float64 weight
